lib/_stream_writable.js

```diff
@@ -107,6 +112,13 @@ function WritableState(options, stream) {
 
   // True if the error was already emitted and should not be thrown again
   this.errorEmitted = false;
+
+  // when `state._end()` was called
+  this.flushed = false;
+
+  // NOTE: added here to not pollute the prototype of the WritableStream and to
+  // avoid conflicts with user-land methods
+  this._end = options._end || _end;
```

Comment:
we could alternatively add `_end` to the `WritableState.prototype` since we don't actually used `_end` for anything else
```javascript
WritableState.prototype.end = function writableStateEnd(stream, cb) {
  cb(null);
};
```
Then here we could just add:
```javascript
if (typeof options.end === 'function')
  this.end = options.end;
```